pkg/server/helpers_unix.go

```diff
@@ -141,3 +148,97 @@ func (c *criService) seccompEnabled() bool {
 func openLogFile(path string) (*os.File, error) {
 	return os.OpenFile(path, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0640)
 }
+
+// unmountRecursive unmounts the target and all mounts underneath, starting with
+// the deepest mount first.
+func unmountRecursive(ctx context.Context, target string) error {
+	mounts, err := mount.Self()
+	if err != nil {
+		return err
+	}
+
+	var toUnmount []string
+	for _, m := range mounts {
+		if strings.HasPrefix(m.Mountpoint, target) {
```

Comment:
Does `target` always end in a `/` here? Since this is string comparison I always feel more comfortable doing path cleanup and explicitly checking for equality or cleaned prefix + `/` to ensure it couldn't catch something like `/something` and `/something-variant`.